### Урок 2. Дискретные случайные величины. Закон распределения вероятностей.  Биномиальный закон распределения. Распределение Пуассона

In [1]:
import numpy as np
from math import factorial
from IPython.display import Image

# Перестановки
def permutations(n):
    return int(factorial(n))

# Размещения
def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

# Сочетания
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

# Биномиальный закон. Формула Бернулли
def binomial(n, k , p):
    return (combinations(n, k)) * (p ** k) * ((1 - p) ** (n - k))

# Формула Пуассона
def poisson_proba(k, lambda_):
    return (lambda_ ** k) * (np.exp(-lambda_)) / np.math.factorial(k)

#### Задание 1
Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите `M(X)`, `D(X)`.

Решение: <br>
Пусть X - дискретная случайная величина, равная числу правильных ответов. Она распределена по биномиальному закону с параметрами n = 5 (число вопросов), p = 0.25 (вероятность угадать ответ), поэтому вероятности вычисляются по формуле Бернулли.

In [2]:
#(P = X = 5)
n = 5
k = 5
p = 0.25
binomial(n, k, p)

0.0009765625

In [3]:
#(P = X = 4)
n = 5
k = 4
p = 0.25
binomial(n, k, p)

0.0146484375

In [4]:
#(P = X = 3)
n = 5
k = 3
p = 0.25
binomial(n, k, p)

0.087890625

In [5]:
#(P = X = 2)
n = 5
k = 2
p = 0.25
binomial(n, k, p)

0.263671875

In [6]:
#(P = X = 1)
n = 5
k = 1
p = 0.25
binomial(n, k , p)

0.3955078125

In [7]:
#(P = X = 0)
n = 5
k = 0
p = 0.25
binomial(n, k, p)

0.2373046875

Закон распределения заданной случайной величины: <br>

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X = k)$</th>
        <td>0.237</td>
        <td>0.395</td>
        <td>0.263</td>
        <td>0.087</td>
        <td>0.014</td>
        <td>0.00097</td>
    </tr>
</table>

$M(X) = np$ - математическое ожидание биномиального распределения.

In [8]:
M = 5 * 0.25
M

1.25

$D(X) = np(1-p)$ - дисперсия биномиального распределения.

In [9]:
D = 5 * 0.25 * (1 - 0.25)
D

0.9375

#### Задание 2
Вероятность попадания в цель при одном выстреле равна 0.01. Какова вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10?

Решение: <br>
Пусть случайная величина X – число попаданий в цель. Так как вероятность p = 0.01 очень мала, а число выстрелов (опытов) достаточно велико, то искомую вероятность будем находить, используя формулу Пуассона. Так как события несовместные, искомая вероятность может быть найдена по формуле сложения вероятностей.

In [10]:
lambda_ = 200 * 0.01
lambda_

2.0

In [11]:
sum(poisson_proba(k = i, lambda_ = lambda_) for i in range(5, 11))

0.052644709119342686

#### Задание 3
В условиях предыдущей задачи сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание?

_Подсказка_. Как посчитать `P(X >= k)` для величины `X`, имеющей распределение Пуассона?

Решение: <br>
P(X>=k) для велечины X равна 1 - P(X< k), следовательно количество промахов при 10 попаданиях будет n - 10. <br>
Значит, наша вероятность равна 0.9 = 1 - p в степени (n - 10). Найдём n путём подбора.

In [12]:
p = 0.1

for i in range(1, 1500):
    n = i
    q = 0.99 ** n
    if 0 < (p - q) < 0.001:
        break
        
print(f'Количество выстрелов равно: {n + 10}')

Количество выстрелов равно: 240
